synapse/http/server.py

```diff
@@ -275,6 +275,10 @@ class DirectServeJsonResource(_AsyncResource):
     formatting responses and errors as JSON.
     """
 
+    def __init__(self, canonical_json=False, extract_context=False):
```

Comment:
I've set the default for `canonical_json` to be False here, whereas it is True for `JsonResource`, simply because I think you want non-canonical-json more often than you want canonical-json. I don't really know why `JsonResource` makes the default True.